pkg/util/extprom/extprom.go

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/thanos-io/thanos/blob/main/pkg/extprom/extprom.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Thanos Authors.
+
+package extprom
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// WrapRegistererWithPrefix is like prometheus.WrapRegistererWithPrefix but it passes nil straight through
+// which allows nil check.
+func WrapRegistererWithPrefix(prefix string, reg prometheus.Registerer) prometheus.Registerer {
```

Comment:
I think we don't need this and `WrapRegistererWithPrefix` anymore, because the upstream `prometheus.Wrap*()` functions now handle a `nil` registerer.